installer.sh

```diff
@@ -49,22 +38,64 @@ echo ""
 echo "Testing if the script works"
 echo ""
 
-timeout 2.5s ./venv/bin/python ./main.py
+# Run the Python script in the background
+./venv/bin/python ./main.py &
+# Get the background process's PID
+PYTHON_PID=$!
+# Sleep for 2.5 seconds
+sleep 2.5
+# Kill the background process
+kill $PYTHON_PID
 
 echo "Test might've worked, did it spit out any errors?"
 echo "Commands executed."
 
-# Turn steam-presence/bin/python into steam-presence/venv/bin/python and replace "/home/deck" with current directory in steam-presence.service file
-echo "Setting up service file"
-mkdir -p "$HOME/.config/systemd/user"
-sed -e "s~steam-presence/bin/python~steam-presence/venv/bin/python~g" -e "s~/home/deck/steam-presence~$PWD~g" "$PWD/steam-presence.service" > $HOME/.config/systemd/user/steam-presence.service
-
+# Create a shell script wrapper for macOS
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  echo '#!/bin/bash' > steam-presence
+  echo "DIR=\"\$( cd \"\$( dirname \"\${BASH_SOURCE[0]}\" )\" && pwd )\"" >> steam-presence
+  echo "exec \"\$DIR/venv/bin/python\" \"\$DIR/main.py\"" >> steam-presence
+  chmod +x steam-presence
+fi
 
-echo "Starting service"
-systemctl --user daemon-reload
-systemctl --user --now enable "steam-presence.service"
+# Detect OS and perform OS-specific tasks
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
```

Comment:
I've never seen this environment variable before?

Reply:
https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html
It is in the Bash documentation. Works on macOS but I don't have a linux env to test it rn.